students/JerryH/Lesson04/mailroom2.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+
+# List of sample donors and donations
+donors = {
+    "Bill Gates": [234.22, 45645.24, 43953.09, 98823],
+    'Jeff Bezo': [4564.23],
+    'Mike Dell': [299.09, 26273.67],
+    'Harry Potter': [8234.09, 83948.04, 7834.23],
+    'Ben Williams': [29283.00, 1334.34],
+    'Guy James': [93.00, 34.34]
+    }
+
+options = {
+    'THANK_YOU_OPT': 1,
+    'REPORT_OPT': 2,
+    'SEND_LETTER_OPT': 3,
+    'QUIT_OPT': 4
+    }
+
+menu = {
+    'op1': "Send a Thank You",
+    'op2': "Create a Report",
+    'op3': "Send Letters To Everyone",
+    'op4': "Quit"
+    }
+
+
+def prompt():
+    return int(input("Please choose the following options:\n1) {op1}.\n2) {op2}.\n3) {op3}.\n4) {op4}\n".format(**menu)))
+
+
+def donor_name_prompt():
+    return input("Send a Thank You - Please enter a full name or type \"list\""
+        "to list the current donors:\n")
+
+
+def get_all_donor_names():
+    all_names = []
+    for d in donors.keys():
+        all_names.append(d)
+    return all_names
```

Comment:
dict is a great choice for `donors` data structure.. what's even better is it is so easy now to get a list of donor names 🙂 
```
>>> list(donors)
['Bill Gates', 'Jeff Bezo', 'Mike Dell', 'Harry Potter', 'Ben Williams', 'Guy James']
>>> list(donors.keys())
['Bill Gates', 'Jeff Bezo', 'Mike Dell', 'Harry Potter', 'Ben Williams', 'Guy James']
>>> donors.keys()
dict_keys(['Bill Gates', 'Jeff Bezo', 'Mike Dell', 'Harry Potter', 'Ben Williams', 'Guy James'])
```
so your function could simply be:
```
def get_all_donor_names():
    return list(donors)
```